accounts-db/src/append_vec.rs

```diff
@@ -85,7 +87,23 @@ pub enum AppendVecError {
 /// A slice whose contents are known to be valid.
 /// The slice contains no undefined bytes.
 #[derive(Debug, Copy, Clone)]
-struct ValidSlice<'a>(&'a [u8]);
+pub(crate) struct ValidSlice<'a>(&'a [u8]);
```

Comment:
When can a slice be invalid?

Reply:
mmap[] of a file that is mmapped in. If the file len is 4096 bytes (becauese of recycling and page alignment), but only 100 bytes are valid, then we'd want a `ValidSlice` that is limited to only the VALID bytes of a potentially larger slice which is memory-wise valid but has garbage data. The introduction of this type allows the fns that take it to know that ALL bytes in the slice are known to be valid. Otherwise, a single caller that didn't pass in `&self.mem[..self.len()]` and instead passed `self.mem` (where mem: &[u8] and represents the ENTIRE file mmapped in). This forces callers to truncate if necessary and allows fns to know all bytes in the slice are valid.